source/common/http/filter_manager.h

```diff
@@ -280,6 +280,10 @@ struct ActiveStreamDecoderFilter : public ActiveStreamFilterBase,
 
   void requestDataTooLarge();
   void requestDataDrained();
+  // Encoding end_stream by a non-terminal filters (i.e. cache filter) always causes the decoding to
+  // be stopped even if independent half-close is enabled. For simplicity, independent half-close is
+  // enabled only when the terminal (router) filter is encoding the response.
```

Comment:
Is this dependent on the router filter itself or any arbitrary terminal filter?

Reply:
I have removed reference to router